engine/framework/model/typedefs.go

```diff
@@ -27,3 +32,82 @@ type (
 	// Epoch is an increasing only value.
 	Epoch = int64
 )
+
+// Defines all task type
+// TODO: Refine me.Currently, when adding a new worker type or job type, we need to modify many code places,
+// NOTICE: DO NOT CHANGE the previous worker type
+// Modify the comment in model IF you add some new worker type
+const (
+	JobManager = WorkerType(iota + 1)
+	// job master
+	CvsJobMaster
+	FakeJobMaster
+	DMJobMaster
+	CdcJobMaster
+	// task
+	CvsTask
+	FakeTask
+	DmTask
+	CdcTask
+	// worker
+	WorkerDMDump
+	WorkerDMLoad
+	WorkerDMSync
+	// extend the worker type here
+)
+
+var toString = map[WorkerType]string{
+	0:             "",
+	JobManager:    "JobManager",
+	CvsJobMaster:  "CVSJobMaster",
+	FakeJobMaster: "FakeJobMaster",
+	DMJobMaster:   "DMJobMaster",
+	CdcJobMaster:  "CDCJobMaster",
+	CvsTask:       "CVSTask",
+	FakeTask:      "FakeTask",
+	DmTask:        "DMTask",
+	CdcTask:       "CDCTask",
+	WorkerDMDump:  "DMDumpTask",
+	WorkerDMLoad:  "DMLoadTask",
+	WorkerDMSync:  "DMSyncTask",
+}
+
+var toID = map[string]WorkerType{
```

Comment:
Why do we need to maintain maps if we have a map `toString`?
Why not initialize the `toID` map in our `init()` function to avoid maintaining two maps?